src/PageLifecycleProvider.tsx

```diff
@@ -139,13 +139,15 @@ export const PageLifecycleProvider: React.FC<PageLifecycleProviderProps> = ({
             'End loading data',
         )
 
-        if (loadingDataCount.current === 0) {
+        // If pages load data after the page has transitioned to complete, just ignore it
+        if (loadingDataCount.current === 0 && isRouting.current) {
```

Comment:
Only raise the page-load-complete event if the page is routing (currently in transition)